spec/app/ics-028-cross-chain-validation/data_structures.md

```diff
@@ -0,0 +1,294 @@
+    ```typescript
+    interface VSCPacketData {
+      // the id of this VSC
+      id: uint64 
+      // validator updates
+      updates: [ValidatorUpdate]
+      // downtime slash requests acknowledgements, 
+      // i.e., list of validator addresses
+      downtimeSlashAcks: [string]
+    }
+    ```
+- `VSCMaturedPacketData` contains the ID of the VSC that reached maturity, i.e., 
+    ```typescript
+    interface VSCMaturedPacketData {
+      id: uint64 // the id of the VSC that reached maturity
+    }
+    ```
+- `SlashPacketData` contains a request to slash a validator, i.e.,
+  ```typescript
+    interface SlashPacketData {
+      valAddress: string // validator address, i.e., the hash of its public key
+      valPower: int64
+      vscId: uint64
+      downtime: Bool
+    }
+    ```
+> Note that for brevity we use e.g., `VSCPacket` to refer to a packet with `VSCPacketData` as its data.
+
+Packets are acknowledged by the remote side by sending back an `Acknowledgement` that contains either a result (in case of success) or an error (as defined in [ICS 4](../../core/ics-004-channel-and-packet-semantics/README.md#acknowledgement-envelope)). 
+The following acknowledgement types are required by the CCV module:
+```typescript
+type VSCPacketAcknowledgement = VSCPacketSuccess | VSCPacketError;
+type VSCMaturedPacketAcknowledgement = VSCMaturedPacketSuccess | VSCMaturedPacketError;
+type SlashPacketAcknowledgement = SlashPacketSuccess | SlashPacketError;
+type PacketAcknowledgement = PacketSuccess | PacketError; // general ack
+```
+
+## CCV State
+[&uparrow; Back to Outline](#outline)
+
+This section describes the internal state of the CCV module. For simplicity, the state is described by a set of variables; for each variable, both the type and a brief description is provided. In practice, all the state (except for hardcoded constants, e.g., `ProviderPortId`) is stored in a key/value store (KVS). The host state machine provides a KVS interface with three functions, i.e., `get()`, `set()`, and `delete()` (as defined in [ICS 24](../../core/ics-024-host-requirements)).
+
+- `ccvVersion = "ccv-1"` is the CCV expected version. Both the provider and the consumer chains need to agree on this version.
+- `zeroTimeoutHeight = {0,0}` is the `timeoutHeight` (as defined in [ICS 4](../../core/ics-004-channel-and-packet-semantics)) used by CCV for sending packets. Note that CCV uses `ccvTimeoutTimestamp` for sending CCV packets and `transferTimeoutTimestamp` for transferring tokens. 
+- `ccvTimeoutTimestamp: uint64` is the `timeoutTimestamp` (as defined in [ICS 4](../../core/ics-004-channel-and-packet-semantics)) for sending CCV packets. The CCV protocol is responsible of setting `ccvTimeoutTimestamp` such that the *Correct Relayer* assumption is feasible.
+- `transferTimeoutTimestamp: uint64` is the `timeoutTimestamp` (as defined in [ICS 4](../../core/ics-004-channel-and-packet-semantics)) for transferring tokens. 
+
+### State on Provider Chain
```

Comment:
Is there no `State on Consumer Chain` section?

Reply:
Yes there is https://github.com/cosmos/ibc/blob/marius/721-restructure/spec/app/ics-028-cross-chain-validation/data_structures.md#state-on-consumer-chain